src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,97 @@
+import _ from 'lodash';
+import React from 'react';
+import Overlay from '../Overlay/Overlay';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createLucidComponentDefinition }  from '../../util/component-definition';
+
+const boundClassNames = lucidClassNames.bind('&-Dialog');
+
```

Comment:
I'm wary of repeatedly used hard-coded values. Can we add
```
const LARGE = 'large';
const MEDIUM = 'medium';
const SMALL = 'small';
```
here and use those throughout this component?